### PR TITLE
ISLANDORA-2164: Update Scopus API URL to HTTPS version

### PR DESCRIPTION
**Pull request by bondjimbond**
**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-2164)

# What does this Pull Request do?

Updates the Scopus API URL to the current HTTPS version. Required by Elsevier - HTTP to be phased out by end of month.

# What's new?
Just the API endpoint.

# How should this be tested?
- Enable the module, enable the block
- Ingest an object with a DOI that returns Scopus citation badge on the original branch
- Check out this branch, clear cache
- Confirm that badge still appears

# Additional Notes:
Versions of the module without this update will stop working in April 2018.

# Interested parties
@Islandora/7-x-1-x-committers

### CODE REVIEW

File: modules/islandora_scopus/islandora_scopus.module
Comment (DiegoPino) [inline modules/islandora_scopus/islandora_scopus.module]: Good practices and third eye tells me this URL should be exposed to the user to be modified anytime its needed. The idea of forcing people to update by a certain date because our hardcoded URL will de deprecated should be avoided in the future.

**Review by DiegoPino — CHANGES_REQUESTED** [review]

File: modules/islandora_scopus/islandora_scopus.module
Comment (bryjbrown) [reply modules/islandora_scopus/islandora_scopus.module]: @DiegoPino I understand that feeling, but alternatively, its really on Elsevier to keep a stable API up. Forcing people to update because they will deprecate their own API is a massive fail on their part, not ours, and allowing people to alter the URL (variable in an admin menu?) introduces some weird complexity where people might have issues because they have set the wrong URL for the API and the module no longer works.

In a nutshell: I think API URLs *should* be hardcoded, because API URLs shouldn't change. People are gonna have to update badges because of Elsevier's fail anyways, lets not normalize their bad behavior.

^ Strong opinion held loosely, just playing devil's advocate here and opening up a debate on the idea. :)

**Comment by bondjimbond** [comment]
Thanks, @DiegoPino - API Endpoint URL now configurable, with the correct one as default.

**Comment by DiegoPino** [comment]
@bondjimbond side question actually. I noticed that many of your config settings, even when they have a default value have no UI way of resetting them to factory defaults. Have you ever experienced this, like you change a default, it does not work, you no longer remember what the default was and you end digging in code(bad, bad UX)? Wonder if we could talk about that as a community for this release, a practice that would be good to replicate on CLAW since UI is starting there fresh. @rosiel what do you think?

**Review by DiegoPino — APPROVED** [review]

**Comment by DiegoPino** [comment]
24 hours and counting! Nice job @bondjimbond++

File: modules/islandora_scopus/includes/admin.form.inc
Comment (DiegoPino) [inline modules/islandora_scopus/includes/admin.form.inc]: Ups, this is wrongly indented, sorry!

**Review by DiegoPino — CHANGES_REQUESTED** [review]

**Comment by bondjimbond** [comment]
@DiegoPino Good point... I do it the way I do it because I just copy how things have been done in the past and don't know any better. Is there actually a way to reset-to-default? Not necessarily something I'll do here but could do in future.

File: modules/islandora_scopus/includes/admin.form.inc
Comment (bondjimbond) [reply modules/islandora_scopus/includes/admin.form.inc]: Whoops!

**Comment by DiegoPino** [comment]
@bondjimbond true. We have been replicating quite a few things and not always we have the time to reflect if the original way of doing things can be improved. That is Ok, we can't rethink everything every time! But I feel we have a history (unvoluntary of course) of giving people a hard time via UI by expecting they need to know the whereabouts of Islandora's internal clockwork. And well, the idea of UI is to abstract that a bit! Lets discuss this tomorrow on the committers call. Out of scope for this pull of course.

**Comment by bondjimbond** [comment]
@DiegoPino Sounds good. I think Travis is happy, so all good to go?

File: modules/islandora_scopus/islandora_scopus.module
Comment (DiegoPino) [reply modules/islandora_scopus/islandora_scopus.module]: @bryjbrownI totally get your point and I appreciate the role of devil's advocate here + your experience (IR archangel!) on this. The reason I went for this (Issue I saw) is that our folks won't blame Elsevier when stuff stop working because we are the ones providing the UI and the connection and forcing updates is something we have done only for security reasons. A bit again about the role of Islandora in this whole scenario where we rely more and more on external data providers, probably this will play against integrators quite more in the future. There are even some API providers that version their APIs and deprecate old endpoints like /myendpoint/v1/? etc... So, now i have mixed feelings. Like, is fedora an API? and why are we asking people to add their fedora endpoints? I propose we discuss this tomorrow! Worst case for @bondjimbond (sorry man) you would have to rollback a bit to where I started interacting. Probably related to my other comment on how we allow stable fallbacks and permit resetting forms to factory defaults. Thanks @bryjbrown, this is a good discussion!

**Comment by DiegoPino** [comment]
@bondjimbond see https://github.com/Islandora/islandora_badges/pull/15#discussion_r172866965 and follow up. We will discuss some of this tomorrow, in any case, we should be close to 24 hours after that. Stay tunned!

File: modules/islandora_scopus/islandora_scopus.module
Comment (bondjimbond) [reply modules/islandora_scopus/islandora_scopus.module]: @DiegoPino If we have to roll back, then it's just a matter of going back to the first commit to this branch. No biggie.

**Comment by bondjimbond** [comment]
Resolution from Committers' Call: keep as-is, and add the correct URL to the Description for the field.

File: modules/islandora_scopus/includes/admin.form.inc
Comment (whikloj) [inline modules/islandora_scopus/includes/admin.form.inc]: HAHAHA, I think your `Ctrl-C + Ctrl-V` went arry.

Comment (bondjimbond) [reply modules/islandora_scopus/includes/admin.form.inc]: Whoops, yer right... sigh

**Comment by bondjimbond** [comment]
@DiegoPino Since it was discussed at the Committers' Call yesterday and updated as requested, are we ready to merge?

**Review by DiegoPino — APPROVED** [review]

**Comment by DiegoPino** [comment]
@bondjimbond will merge during the day. thanks. @bryjbrown could you prepare some awesome public statement for next week to let people know of this needed update and something we can also add to the release notes? http will be deprecated before release so good to go public on that as soon as this gets merged (monday or tuesday). Thanks team

**Comment by bryjbrown** [comment]
@DiegoPino I've added it to the agenda for the next IRIG meeting, but @bondjimbond should get to be the one to make the announcement since he knows the most about it and did all the hard work, I wouldn't want to steal his thunder! All glory to @bondjimbond! :1st_place_medal:

**Comment by DiegoPino** [comment]
Ja! True, but since @bryjbrown is IR Supreme leader (and Sheriff)  and also the maintainer of this difficult child Scholar, it could be nice if you both go public, maybe @bondjimbond explaining the work (CC) and you pushing people into Update mode? I mean, yes, glory to @bondjimbond too of course but someone needs to convince folks out there to update before hell breaks loose and we know IR users trust you with their lives! 😄  I let you both decide on this of course.  Good work

**Comment by bondjimbond** [comment]
@DiegoPino This one's still waiting for a merge. Could you please hit the button? Afterward I'll draft up an announcement about the update for the Google group.

**Comment by DiegoPino** [comment]
@bondjimbond fair. My day job implies more than Islandora and i get derailed. Merging
As a side note for your public release: changing this to `https` does not make SSL certs verifications happen or adds any other security improvements on our side. See https://www.drupal.org/node/1081192

**Comment by bondjimbond** [comment]
Thanks, @DiegoPino!

**Pull request merged**
